frontend/app/.server/domain/services/city-service-default.ts

```diff
@@ -72,17 +80,22 @@ export const cityService: CityService = {
    */
   async getByCode(code: string): Promise<Result<City, AppError>> {
     const context = `get city with CODE '${code}'`;
-    try {
-      const response = await apiFetch(`/cities?code=${code}`, context);
-      const data: City = await response.json();
-      return Ok(data);
-    } catch (error) {
-      if (error instanceof AppError && error.httpStatusCode === HttpStatusCodes.NOT_FOUND) {
-        return Err(new AppError(`City with CODE '${code}' not found.`, ErrorCodes.NO_CITY_FOUND));
-      }
-      // Re-throw any other error
-      throw error;
+    type ApiResponse = {
+      content: readonly City[];
+    };
+    const response = await apiClient.get<ApiResponse>(`/cities?code=${code}`, context);
+    if (response.isErr()) {
+      throw response.unwrapErr();
     }
+    const data = response.unwrap();
+    const city = data.content[0]; // Get the first element from the response array
```

Comment:
`Array.prototype.at()` (ES2022)

```suggestion
    const city = data.content.at(0); // Get the first element from the response array
```

or

Destructuring Assignment

```suggestion
    const [city] = data.content; // Get the first element from the response array
```